core/cap-0067.md

```diff
@@ -172,7 +188,7 @@ Emit one of the following events -
 
 For a payment not involving the issuer, or if both the sender and receiver are the issuer:
 ```
-contract: asset, topics: ["transfer", from:Address, to:Address, sep0011_asset:String], data: amount:i128
+contract: asset, topics: ["transfer", from:Address, to:Address, sep0011_asset:String], data: { amount:i128 }
```

Comment:
I'm not sure what is this supposed to mean, given that for all the remaining fields the names do not exist in the events. 

We need to update SEP-41, but given that a singular i128 field in the data has to be supported for the historical purposes, I guess that the spec would say 'either i128 field with transfer amount, or a map that has at least `amount(Symbol)->amount_value:i128` KV pair present. Then we could emit i128 unless we also emit mux ids.

I assume that if we had balance in the events, then we'd need to always emit maps, but we don't seem to do that for p23.

Reply:
`{ }` is a map. `amount` is an assumed Symbol. Consistent I think what you're saying. We should be more explicit though. 👍🏻

Reply:
Yeah my intention was to say the data field was always a map, so even in the `data: amount:i128` case, you'd see `amount(Symbol)->amount_value:i128`. I'll update the CAP to make this clear.

Dima IIUC, I think you're actually advocating for still emitting the data field as an integer today if we're emitting only the amount because we'll always have to support that anyways?

Reply:
Yeah my intention was to say the data field was always a map, so even in the `data: amount:i128` case, you'd see `amount(Symbol)->amount_value:i128`. I'll update the CAP to make this clear.
@dmkozh IIUC, I think you're actually advocating for still emitting the data field as an integer today if we're emitting only the amount because we'll always have to support that anyways?

Reply:
Emitting a map for all is a way to ensure all downstream systems see this change and aren't surprised by the first time they encounter the edge cases like muxed accounts.

Reply:
> We should be more explicit though.

Yeah, there are no implied symbols here otherwise. Also, some events were not updated, but I assume that was unintentional.

> Emitting a map for all is a way to ensure all downstream systems see this change and aren't surprised by the first time they encounter the edge cases like muxed accounts.

This is kind of debatable, since the integer format still has to be supported and map costs more to emit. If someone wanted to integrate with the unified events, they'd need to support both formats either way, and it's not too hard to discover. That said, I don't feel about this too strongly; I can see some merit to using the map (almost) everywhere as well.

Reply:
👍🏻 I agree with all those points. Especially if other events won't be updated and will still emit a u64 data section.

Reply:
I updated the CAP to use an integer data field if only the amount is being emitted, and a map otherwise with some details added.